Lib/glyphsLib/builder/anchors.py

```diff
@@ -43,13 +43,19 @@ def _propagate_glyph_anchors(ufo, parent, processed):
     anchor_names = set()
     to_add = {}
     for component in parent.components:
-        glyph = ufo[component.baseGlyph]
-        _propagate_glyph_anchors(ufo, glyph, processed)
-        if any(a.name.startswith('_') for a in glyph.anchors):
-            mark_components.append(component)
+        try:
+            glyph = ufo[component.baseGlyph]
+        except KeyError:
+            self.logger.warn(
```

Comment:
`warn` method is deprecated, we should use `logger.warning`
https://docs.python.org/3/library/logging.html#logging.warning